### ISSUE TITLE
Add eye tracking support for XR SDK

### ISSUE THREAD

**Issue by keveleigh** [microsoft/MixedRealityToolkit-Unity#7284]
## Describe the problem

XR SDK support in MRTK 2.3 is shipping without support for the XR SDK eye tracking subsystem. Using our existing eye tracking profiles, we should add support on the new XR pipeline as well.

## Additional context

Follow-up issue to #7003

**Comment by deibu** [comment]
Any update on when this will be added?

**Comment by deibu** [comment]
Was eye tracking support added in MRTK 2.4.0? I am using the XR SDK and eye tracking seems to be working.

**Comment by keveleigh** [comment]
@deibu Huh, I don't remember us adding support in 2.4 via XR SDK, no. It's one of the things upcoming on my plate. Not sure what's happening there! Is this on a HoloLens 2?

**Comment by deibu** [comment]
> @deibu Huh, I don't remember us adding support in 2.4 via XR SDK, no. It's one of the things upcoming on my plate. Not sure what's happening there! Is this on a HoloLens 2?

Yes, I am using the HoloLens 2 with MRTK 2.4.0, XR SDK and Unity 2019.4.5f1. I thought it wasn't working for the longest time, but then noticed that the gaze cursor was active at the beginning of running my application (if my hands were not being tracked) and was in fact following the eye tracking. The gaze cursor quickly disappears as hands become active, and then never become active again with the "Select" voice command (#8582). If I use PointerUtils.SetGazePointerBehavior(PointerBehavior.AlwaysOn), then the gaze cursor remains active and eye tracking works as expected.

**Comment by yacuzo** [comment]
Is this still not in MRTK 2.5? I can't seem to find a EyeGazeDataProvider for the XR SDK. Should it work if we implement one, or is  there other things needed?

**Comment by deibu** [comment]
@yacuzo In MRTK 2.4.0, I was able to use the WindowsMixedRealityEyeGazeDataProvider with XR SDK. I don't know if it's actually supposed to work with XR SDK, but it did in my project.

**Comment by keith-phillips** [comment]
Hi @keveleigh 

Is there any news on this? We're also using the HoloLens 2 and this is preventing us from upgrading to Unity 2020 since we need eye tracking. Unfortunately we also really need a bug fix that has gone into Unity 2020.1!

Are there any suggested short-term solutions to enable use of the MRTK with eye tracking in Unity 2020? We could look at plumbing through the XRSDK data ourselves, but it looks from issue #8715 that this isn't going to work either...?

**Comment by 9and3** [comment]
Also looking for making the Eye-gazing working with the XRSDK!

**Comment by ModischFabrications** [comment]
Currently doing my masters thesis and ran into the same issue. Using Unity 2020, where only XR SDK is supported, so there isn't a simple way to downgrade to legacy. Did you find a workaround to access Eye Gaze with XR SDK @keith-phillips? Don't really want to downgrade to Unity 2019 and change all components to outdated legacy code.

**Comment by deibu** [comment]
> Currently doing my masters thesis and ran into the same issue. Using Unity 2020, where only XR SDK is supported, so there isn't a simple way to downgrade to legacy. Did you find a workaround to access Eye Gaze with XR SDK @keith-phillips? Don't really want to downgrade to Unity 2019 and change all components to outdated legacy code.

If you're okay with manually turning it on, you should be able to force it with PointerUtils.SetGazePointerBehavior(PointerBehavior.AlwaysOn).

**Comment by ModischFabrications** [comment]
> PointerUtils.SetGazePointerBehavior(PointerBehavior.AlwaysOn)

Just tried it out in a few combinations, looks like it enables the head gaze instead of the eye gaze. Is there something else you enabled or added? Did you add WindowsMixedRealityEyeGazeDataProvider somehow or did you skip it completely? I am unable to select it as an input data provider. 

Setup: 
MRTK 2.5.4, `Use Head Gaze override = true`, `Is eye tracking enabled = true` in pointer settings. `IsEyeTrackingEnabledAndValid` still returns false.

**Comment by deibu** [comment]
> > PointerUtils.SetGazePointerBehavior(PointerBehavior.AlwaysOn)
> 
> Just tried it out in a few combinations, looks like it enables the head gaze instead of the eye gaze. Is there something else you enabled or added? Did you add WindowsMixedRealityEyeGazeDataProvider somehow or did you skip it completely? I am unable to select it as an input data provider.
> 
> Setup:
> MRTK 2.5.4, `Use Head Gaze override = true`, `Is eye tracking enabled = true` in pointer settings. `IsEyeTrackingEnabledAndValid` still returns false.

I followed the Getting Started guide (https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/EyeTracking/EyeTracking_BasicSetup.html).

I did add the WindowsMixedRealityEyeGazeDataProvider to my custom XRSDKInputSystemProfile. You'll also want to verify that the built VS project has gazeinput in the Package.appxmanifest file under Capabilities.

**Comment by ModischFabrications** [comment]
> I did add the WindowsMixedRealityEyeGazeDataProvider to my custom XRSDKInputSystemProfile

How? That's the big problem, it's just not selectable as an input provider in any of the folders available to me: 
![image](https://user-images.githubusercontent.com/25404728/108379423-33d68880-7206-11eb-9bf0-a666408b4cea.png)

**Comment by deibu** [comment]
> > I did add the WindowsMixedRealityEyeGazeDataProvider to my custom XRSDKInputSystemProfile
> 
> How? That's the big problem, it's just not selectable as an input provider in any of the folders available to me:
> ![image](https://user-images.githubusercontent.com/25404728/108379423-33d68880-7206-11eb-9bf0-a666408b4cea.png)

This is where I found it:
![image](https://user-images.githubusercontent.com/2654631/108409396-088a7400-71db-11eb-99c2-cd90770bf970.png)

**Comment by ModischFabrications** [comment]
I am very thankful for your help, but something does not match up here. I don't have that selection, even *.WindowsMixedReality.* is gone now, are you on MRTK 2.5.4 as well or still on 2.4?

**Comment by deibu** [comment]
> I am very thankful for your help, but something does not match up here. I don't have that selection, even _.WindowsMixedReality._ is gone now, are you on MRTK 2.5.4 as well or still on 2.4?

MRTK 2.5.1
Windows XR Plugin 2.4.1
XR Plugin Management 3.2.16

**Comment by keveleigh** [comment]
Moving this out a release, since we're waiting on some fixes from Unity (in review)
